### PR TITLE
add details about underlying message types on input message protocol errors

### PR DESCRIPTION
**Pull request by tmm1**
in general the protocol errors are all missing a lot of context

### CODE REVIEW

**Comment by srikrsna-buf** [comment]
Hey @tmm1 Can you tell us a bit about how often you encountered protocol errors and what the backend was? Because we don't expect users to ever hit these errors.

**Comment by tmm1** [comment]
We are seeing large numbers of these proto errors and are trying to figure out why. Adding these details here would help.

File: packages/connect/src/protocol/invoke-implementation.ts [hunk 1]
```diff
@@ -149,8 +149,16 @@ export function transformInvokeImplementation<
...
+            `protocol error: received extra input message ${spec.method.I.typeName} for unary method ${spec.method.name}`,
             Code.Unimplemented,
+            {
+              methodName: spec.method.name,
+              serviceTypeName: spec.service.typeName,
+              inputTypeName: spec.method.I.typeName,
+              outputTypeName: spec.method.O.typeName,
+              url: context.url,
+              requestMethod: context.requestMethod,
+            }
```
Comment (srikrsna-buf) [inline packages/connect/src/protocol/invoke-implementation.ts:161]: Metadata is used for errors returned as part of the protocol, fully qualified method name should be sufficient to pin point the rpc

File: packages/connect/src/protocol/invoke-implementation.ts [hunk 2]
```diff
@@ -149,8 +149,16 @@ export function transformInvokeImplementation<
         const input2 = await inputIt.next();
         if (input2.done !== true) {
           throw new ConnectError(
-            "protocol error: received extra input message for unary method",
+            `protocol error: received extra input message ${spec.method.I.typeName} for unary method ${spec.method.name}`,
```
Comment (srikrsna-buf) [inline packages/connect/src/protocol/invoke-implementation.ts:152]: Fully qualified service name is unique and will be sufficient to pinpoint:
```suggestion
            `protocol error: received extra input message for unary method ${spec.service.typeName}.${spec.method.name}`,
```

**Comment by srikrsna-buf** [comment]
Hey @tmm1 Can you tell us what is the client in use?

File: packages/connect/src/protocol/invoke-implementation.ts (hunk 1 above)
Comment (tmm1) [reply packages/connect/src/protocol/invoke-implementation.ts:161]: If I understand correctly, you are suggesting I remove this as adding metadata would interfere with other parts of the stack?

I can do that. I agree that the rpc method is the main piece of information missing.

**Comment by tmm1** [comment]
We're using typescript on both sides.

**Comment by tmm1** [comment]
Thanks for the guidance. I believe this may be an application specific issue, so I will close for now.

**Pull request closed without merging**